### PR TITLE
View Campaign List Page

### LINKED ISSUE #22: View campaign list page
**Description of feature/enhancement**
You can use the CampaignListItem component that will be added in an upcoming PR.
Screen Shot 2022-12-11 at 10 21 49 PM

_insert link to figma etc_

### PR DESCRIPTION
**Pull request by Enigmatrix**
Closes #22.

This requires #25 to be merged first, then this branch will be rebased on top of it.

**Overview of changes:**

Added mui data tables and added the donations page. Campaign model is defined in `/models`.

**Anything you'd like to highlight/discuss:**

Routing to campaign and retrieving items is not done yet. The page also relies on test data.

### CODE REVIEW

File: models/campaign.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  GridColDef,
+  GridRowParams,
+} from '@mui/x-data-grid'
+
+// GridActionsCellItem seems to be badly defined, I can't set the
+// properties described here: https://mui.com/x/react-data-grid/column-definition/#special-properties
+// So I've created this monstrosity. Forgive me.
+const UntypesafeBox = (props: any) => {
+  return <GridActionsCellItem {...props} />
+}
```
Comment (wilinetan) [inline models/campaign.tsx:14]: rip i also tried but didn't work. their examples on their docs also just used it like this

File: pages/campaigns/donations.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        <Searchbar search={search} setSearch={setSearch} filters={filters} />
+      </Box>
+      <Box sx={{ flex: 1 }}>
+        <DocumentListTabs status={status} setStatus={setStatus}>
+          <DataGrid
+            columns={campaignColumns(handleView)}
+            rows={campaigns}
+            autoHeight
+            pageSize={10}
+          />
```
Comment (wilinetan) [inline pages/campaigns/donations.tsx:81]: can think about how you would display the data for the diff tabs

Comment (Enigmatrix) [reply pages/campaigns/donations.tsx:81]: I thought it was to just use the tabs as a filter as such: https://github.com/dsc-project-mr/mr-fe-internal/blob/19337966cd8b23844e29e02522d71d7834984314/pages/campaigns/donations.tsx#L53-L57

I'm assuming the filtering is done server-side but we provide the prop filters.

Comment (wilinetan) [reply pages/campaigns/donations.tsx:81]: ohh i was thinking like if we do it server side should we still like hold the data for the diff tabs too? so we dont have to keep calling the apis everytime we switch tabs

**Review by wilinetan — APPROVED** [review]

**Pull request merged**